src/gui/tray/Window.qml

```diff
@@ -139,6 +146,92 @@ ApplicationWindow {
         }
     }
 
+    Drawer {
+        id: fileDetailsDrawer
+        width: parent.width
+        height: parent.height
+        padding: 0
+        edge: Qt.RightEdge
+        modal: false
+        visible: false
+
+        background: Rectangle {
+            radius: Systray.useNormalWindow ? 0.0 : Style.trayWindowRadius
+            border.width: Style.trayWindowBorderWidth
+            border.color: Style.menuBorder
+            color: Style.backgroundColor
+        }
+
+        property var folderAccountState: ({})
+        property string fileLocalPath: ""
+        property var pageToShow: Systray.FileDetailsPage.Activity
+
+        function openFileDetails(accountState, localPath, fileDetailsPage) {
+            console.log(`About to show file details view in tray for ${localPath}`);
+            folderAccountState = accountState;
+            fileLocalPath = localPath;
+            pageToShow = fileDetailsPage;
+
+            if(!opened) {
+                open();
+            }
+        }
+
+        Loader {
+            id: fileDetailsContents
+            anchors.fill: parent
+            active: fileDetailsDrawer.visible
+            onActiveChanged: {
+                if (active) {
+                    Systray.showFileDetailsPage(fileDetailsDrawer.fileLocalPath,
+                                                fileDetailsDrawer.pageToShow);
+                }
+            }
+            sourceComponent: ColumnLayout {
+                anchors.fill: parent
+
+                FileDetailsPage {
+                    id: fileDetails
+
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+
+                    background: null
+                    accountState: fileDetailsDrawer.folderAccountState
+                    localPath: fileDetailsDrawer.fileLocalPath
+                }
+
+                CustomButton {
+                    FontMetrics {
+                        id: doneButtonFm
+                        font: parent.contentsFont
+                    }
+
+                    Layout.alignment: Qt.AlignRight | Qt.AlignVCenter
+                    Layout.topMargin: fileDetails.intendedPadding
+                    Layout.bottomMargin: fileDetails.intendedPadding
+                    Layout.leftMargin: fileDetails.intendedPadding
+                    Layout.rightMargin: fileDetails.intendedPadding
+                    Layout.preferredWidth: doneButtonFm.boundingRect(text).width +
+                                           leftPadding +
+                                           rightPadding +
+                                           Style.standardSpacing * 2
+
+                    text: qsTr("Done")
+                    contentsFont.bold: true

```

Comment:
this is not yet defined

Reply:
This PR depends on https://github.com/nextcloud/desktop/pull/5129 (I wrote it in the PR description but it is easy to miss)